Add GDI 05c

reaperrr opened the pull request:
Well, 05eb from the original really, but until we implement campaign progression, calling it 'c' is the easiest way to differentiate it without breaking our mission title length/alignment conventions.

Lua script largely copied from our 05b, with some adjustments where necessary.

Some notes:
- I had to guess the mission selection arrow, so it's possible I accidentally chose the one for 'ea'. Since we're using the wa/wb arrows for the other two missions, I think it wouldn't be a showstopper if I chose the wrong one, but it wouldn't be that much work to fix it if anyone insists.
- The delays for the initial one-time Nod attacks are based on the original (20, 40, 60, 80), but seem a bit short considering how far away the damaged base is from the player starting position, so depending on feedback maybe we should increase the delays some more.
- I limited difficulty differences to the attack delays, repair threshold and a few initial Nod actors, since the inability to sell sandbags and the better Nod units vs. the original (in particular rocket soldiers) already make this mission harder than it was originally.

reaperrr committed:
Add GDI 05c

Well, 05eb really, but until we implement campaign progression,
calling it 'c' is the easiest way to differentiate it without breaking
our mission title length/alignment conventions.

Smittytron commented on mods/cnc/maps/gdi05c/gdi05c.lua:
```diff
@@ -0,0 +1,162 @@
+	end
+
+	--Spawn harv only when base is discovered, otherwise it will waste tiberium and might get killed before the player arrives
+	InsertGDIHarvester()
```
I'd just put the actor spawn line here to avoid a single line function.

Smittytron commented on mods/cnc/maps/gdi05c/gdi05c.lua:
```diff
@@ -0,0 +1,162 @@
+end
+
+WorldLoaded = function()
+	
```
Remove space.

Smittytron commented on mods/cnc/maps/gdi05c/gdi05c.lua:
```diff
@@ -0,0 +1,162 @@
+	hard = { },
+}
+
+AllToHuntTrigger =
```
`AllToHuntTrigger = { Silo1, Proc1, Silo2, Radar1, Afld1, Hand1, Nuke1, Nuke2, Nuke3, Fact1 }`

Smittytron commented on mods/cnc/maps/gdi05c/gdi05c.lua:
```diff
@@ -0,0 +1,162 @@
+
+	Trigger.AfterDelay(AutoAtkStartDelay[Difficulty], AutoCreateTeam)
+
+	Trigger.OnAllRemovedFromWorld(AllToHuntTrigger, function()
```
Use `OnAllKilledOrCaptured` here.

Smittytron commented on mods/cnc/maps/gdi05c/map.yaml:
```diff
@@ -0,0 +1,768 @@
+Actors:
+	Actor0: sbag
+		Location: 62,34
+		Owner: Neutral
```
It's a pain but walls need to be assigned a correct owner so the enemy player can target them.

reaperrr commented:
Updated.

Smittytron commented on mods/cnc/maps/gdi05c/rules.yaml:
```diff
@@ -0,0 +1,169 @@
+E3:
+	Buildable:
+		Prerequisites: ~disabled
+
```
E5 (chem trooper) needs production disabled as well.

Smittytron reviewed (commented):
LGTM otherwise. We may want to revisit the set of GDI05 later for some general responsiveness improvements but this is in line with the other two for now.

Smittytron commented on mods/cnc/maps/gdi05c/rules.yaml:
```diff
@@ -0,0 +1,169 @@
+			easy: Easy
+			normal: Normal
+			hard: Hard
+		Default: easy
```
Default should be normal if we're wanting this to be standardized across missions. (#19426)

reaperrr commented:
Updated.

obrakmann commented on mods/cnc/maps/gdi05c/gdi05c.lua:
```diff
@@ -0,0 +1,154 @@
+	end
+
+	--Spawn harv only when base is discovered, otherwise it will waste tiberium and might get killed before the player arrives
+	Reinforcements.Reinforce(GDI, GDIHarvester, { GdiProc1.Location + CVec.New(0, 2), GdiProc1.Location + CVec.New(-1, 3) }, 1)
```
Happens when you're too slow to discover your base:
```
Fatal Lua Error: [string "-- Make copies of global functions to preve..."]:30: Actor 'proc (dead)' does not define a property 'Location'
   at OpenRA.Scripting.ScriptContext.FatalError(String message)
   at OpenRA.Mods.Common.Scripting.ScriptTriggers.OpenRA.Mods.Common.Traits.INotifyDiscovered.OnDiscovered(Actor self, Player discoverer, Boolean playNotification)
```

obrakmann reviewed (dismissed):
Looks fine otherwise. IIRC correctly, the other gdi05s also had the problem that the enemy infantry bunched up in front of the Hand of Nod, so I'll let that slide.

obrakmann commented on mods/cnc/maps/gdi05c/gdi05c.lua:
```diff
@@ -0,0 +1,154 @@
+
+	InitObjectives(GDI)
+
+	RepairNamedActors(Nod, RepairThreshold[Difficulty])
```
Oh, and you're missing a `Difficulty = Map.LobbyOption("difficulty")`. Interestingly, this doesn't cause a crash on your branch, but does when it gets rebased onto bleed.

abcdefg30 replied:
Yes, because that line was removed from the global script...

reaperrr commented:
> Happens when you're too slow to discover your base:

Added an `if not (...).IsDead` safeguard.

> Oh, and you're missing a Difficulty = Map.LobbyOption("difficulty")

Rebased and added.

obrakmann reviewed (approved):
:+1:

obrakmann commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/196592cd53946f2c7d29da78defe6f24a1ec7af6)
